azext_iot/central/services/api_token.py

```diff
@@ -0,0 +1,125 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import requests
+
+from knack.log import get_logger
+from azext_iot.constants import CENTRAL_ENDPOINT
+from azext_iot.central.services import _utility
+from azext_iot.central.models.enum import Role
+
+logger = get_logger(__name__)
+
+BASE_PATH = "api/preview/apiTokens"
+
+
+def add_api_token(
+    cmd,
+    app_id: str,
+    token_id: str,
+    role: Role,
+    token: str,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+):
+    """
+    Add an API token to a Central app
+
+    Args:
+        cmd: command passed into az
+        app_id: name of app (used for forming request URL)
+        token_id: Unique ID for the API token.
+        role : permission level to access the application
+        token: (OPTIONAL) authorization token to fetch device details from IoTC.
+            MUST INCLUDE type (e.g. 'SharedAccessToken ...', 'Bearer ...')
+        central_dns_suffix: {centralDnsSuffixInPath} as found in docs
+
+    Returns:
+    token: dict
+    """
+    url = "https://{}.{}/{}/{}".format(app_id, central_dns_suffix, BASE_PATH, token_id)
```

Comment:
Are there no api versions set for the central service calls?

Reply:
Its in the `BASE_PATH`, we currently only have the `/preview/` version

Reply:
I see, thanks. I hope there is not an inadvertent API behavior change that could break something on clients. For ARM services typically there is an api version query string with an exact version (even for preview) to avoid/reduce integration issues.